### PR TITLE
add createIntersectionObserver for weixin-app

### PR DESCRIPTION
**Pull request by deepcoldy**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://developers.weixin.qq.com/miniprogram/dev/api/intersection-observer.html#wxcreateintersectionobserverthisoptions>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by AlexStacker — APPROVED** [review]

File: types/weixin-app/index.d.ts [hunk 1]
```diff
@@ -3166,6 +3166,56 @@ declare namespace wx {
...
+        };
+        relativeRect: {
+            bottom: number;
+            right: number;
+            left: number;
+            top: number;
+        };
+    }
+    interface createIntersectionObserverCallBack {
+        relativeTo: (selector?: string, margins?: margins) => createIntersectionObserverCallBack;
```
Comment (taoqf) [inline types/weixin-app/index.d.ts]: try this:
```ts
relativeTo(selector?: string, margins?: margins): createIntersectionObserverCallBack;
```

**Review by taoqf — CHANGES_REQUESTED** [review]

**Review by taoqf — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Can you resolve the merge conflicts? Sorry!

File: types/weixin-app/index.d.ts [hunk 2]
```diff
@@ -3166,6 +3166,56 @@ declare namespace wx {
         touches: Touch[];
         changedTouches: Touch[];
     }
+
+    /**
+     * WXML节点布局相交状态
+     */
+    interface createIntersectionObserverOption {
```
Comment (Jimexist) [inline types/weixin-app/index.d.ts]: upper camel case?

File: types/weixin-app/index.d.ts [hunk 3]
```diff
@@ -3166,6 +3166,56 @@ declare namespace wx {
...
+        thresholds?: number[];
+        initialRatio?: number;
+        selectAll?: boolean;
+    }
+    interface margins {
```
Comment (Jimexist) [inline types/weixin-app/index.d.ts]: upper camel case?

File: types/weixin-app/index.d.ts [hunk 4]
```diff
@@ -3166,6 +3166,56 @@ declare namespace wx {
...
+        left?: number;
+        right?: number;
+        top?: number;
+        bottom?: number;
+    }
+    interface observeResponse {
```
Comment (Jimexist) [inline types/weixin-app/index.d.ts]: upper camel case?

File: types/weixin-app/index.d.ts [hunk 5]
```diff
@@ -3166,6 +3166,56 @@ declare namespace wx {
...
+            height: number; // 相交区域的高度
+        };
+        relativeRect: {
+            bottom: number;
+            right: number;
+            left: number;
+            top: number;
+        };
+    }
+    interface createIntersectionObserverCallBack {
```
Comment (Jimexist) [inline types/weixin-app/index.d.ts]: upper camel case?

File: types/weixin-app/index.d.ts [hunk 6]
```diff
@@ -3166,6 +3166,56 @@ declare namespace wx {
...
+        right?: number;
+        top?: number;
+        bottom?: number;
+    }
+    interface observeResponse {
+        id: string;
+        dataset: any;
+        time: number;
+        intersectionRatio: number; // 相交区域占目标节点的布局区域的比例
+        boundingClientRect: {
```
Comment (Jimexist) [inline types/weixin-app/index.d.ts]: upper camel case?

**Review by jimexist — APPROVED** [review]

**Comment by minestarks** [comment]
@deepcoldy if you could resolve the conflicts I'll merge. Thanks!

**Comment by deepcoldy** [comment]
@minestarks Done. Thanks.

**Pull request merged**